### DISCUSSION TITLE
IDEs using .NET 8.0

### DISCUSSION THREAD

**Discussion by Jillinger** [stride3d/stride#2099] [Q&A] [answered]
Are there any alternate IDEs to Visual Studio, which use .NET 8.0?
JetBrains Rider does, but that's not free. Does SharpDevelop or MonoDevelop support .NET 8.0 SDK?

**Comment by IXLLEGACYIXL** [comment, marked as answer]
VS code works with net8.0 https://code.visualstudio.com/docs/languages/dotnet

VS code has a syntax highlighting extension for stride shaders, but ONLY VS community plugin has also autocompletion

having it side by side installed is always a good idea. If you dont care about shader programming then VS code is completely fine

**Comment by Jillinger** [comment]
Thanks. I just have one problem.
How do I let Stride know I want to use VS Code as the default IDE?
The dropdown in Settings only has two options - Default IDE and Visual Studio Professional 2022.
I edited the GameStudioSetting.conf file in '...AppData\Roaming\Stride\':

> !SettingsFile
> Settings:
>     ExternalTools/DefaultIDE: !file C:/Program Files/Microsoft VS Code/Code.exe

However, no changes are made to the Editor.

**Reply by IXLLEGACYIXL** [reply]
set in Windows the sln file to standard open for VS Code and use in stride "system ide" or something like that
maybe also defaulting csproj needed in windows

**Comment by Jillinger** [comment]
> set in Windows the sln file to standard open for VS Code

I don't want VS Code to be the default program to open .sln files.


> and use in stride "system ide" or something like that

I don't understand. What do you mean?


> maybe also defaulting csproj needed in windows

Visual Studio is set as default for opening these files.
Perhaps the developers at Stride could consider a Settings system that allows users to set paths to IDEs, just as is done for platforms and SDKs.
This is standard for other Engines I have used.

**Reply by IXLLEGACYIXL** [reply]
VS is currently hard coded into Stride, the only way to set it is through setting a default IDE in windows and let stride launch the default IDE

**Reply by Jillinger** [reply]
I don't think you are saying no updates or changes will be made to the Stride Engine.
I don't think you are saying writing a simple code to bypass the default IDE is hard, either.
Perhaps you are just saying, that's the way it is right now.
Thanks for your time.
